model/src/main/proto/feedback_reporting.proto

```diff
@@ -0,0 +1,136 @@
+
+package model;
+
+import "profile.proto";
+
+option java_package = "org.oppia.android.app.model";
+option java_multiple_files = true;
+
+// Top-level proto to store feedback reports locally when there is no network connection to upload
+// them to the remote service
+message FeedbackReportingDatabase {
+  repeated FeedbackReport reports_list = 1;
+}
+
+// Corresponds to an entire feedback report with user reported information and collected app,
+// device, and build information.
+message FeedbackReport {
+  // The date the report was created.
+  int64 report_date_ms = 1;
+  FeedbackReportingForm feedback_form = 2;
+  FeedbackReportingDeviceInfo device_info = 3;
+  FeedbackReportingBuildInfo build_info = 4;
+  FeedbackReportingAppInfo app_info = 5;
+}
+
+// Information supplied by the user in the feedback report.
+message FeedbackReportingForm {
+  oneof report_type {
+    Suggestion suggestion = 1;
+    Issue issue = 2;
+    Crash crash = 3;
+  }
+}
+
+// A suggestion feedback report.
+message Suggestion {
+  enum SuggestionCategory {
+    CATEGORY_UNSPECIFIED = 0;
+    FEATURE_SUGGESTION = 1;
+    LANGUAGE_SUGGESTION = 2;
+    OTHER_SUGGESTION = 3;
+  }
+  SuggestionCategory suggestion_category = 1;
+  // The text input provided by the user.
+  string user_submitted_suggestion = 2;
+}
+
+// An issue feedback report.
+message Issue {
+  enum IssueCategory {
+    CATEGORY_UNSPECIFIED = 0;
+    LESSONS_QUESTIONS_ISSUE = 1;
+    LANGUAGE_ISSUE = 2;
+    LANGUAGE_AUDIO_ISSUE = 3;
+    LANGUAGE_TEXT_ISSUE = 4;
+    TOPICS_ISSUE = 5;
+    PROFILE_ISSUE = 6;
+    OTHER_ISSUE = 7;
+  }
+  IssueCategory issue_category = 1;
+  // List of checkbox items selected by user in the feedback report.
+  repeated string issues_list = 2;
+  // The user's text input if they select "other" for any option
+  string other_user_input = 3;
+}
+
+// A crash report.
+message Crash {
+  enum CrashLocation {
+    LOCATION_UNSPECIFIED = 0;
+    LESSON_PLAYER = 1;
+    PRACTICE_QUESTIONS = 2;
+    TOPICS_LIST = 3;
+    OPTIONS_ACTIVITY = 4;
+    PROFILE_ACTIVITY = 5;
+    OTHER = 6;
+  }
+  CrashLocation crash_location = 1;
+  // The user's text input for explaining what happened.
+  string crash_explanation = 2;
+}
+
+// Information collected about the user's device.
+message FeedbackReportingDeviceInfo {
+  string package_name = 1;
+  string package_code = 2;
+  string country_locale = 3;
+  string language_locale = 4;
+}
+
+// Information collected about the device's build.
+message FeedbackReportingBuildInfo {
+  enum NetworkType {
+    NETWORK_TYPE_UNSPECIFIED = 0;
+    WIFI = 1;
+    CELLULAR = 2;
+  }
+  string device_model = 1;
+  string sdk_version = 2;
+  string device_brand = 3;
+  string build_fingerprint = 4;
+  string phone_type = 5;
+  NetworkType network_type = 6;
+  string device_language = 7;
+}
+
+// Information collected about the app's build.
+message FeedbackReportingAppInfo {
+  enum EntryPoint {
+    ENTRY_POINT_UNSPECIFIED = 0;
+    MAIN_MENU = 2;
```

Comment:
Suggest ``NAVIGATION_DRAWER`` to be consistent with other names for the menu.

Reply:
Done, also added `REVISION_CARD` since feedback reporting can be shown there

Reply:
Ah, what about for questions, too? I'm wondering if we should just focus on explorations & navigation drawer for now. It's easy to add other entry points in the future. What's the main advantage we get for adding it to the revision card toolbar?

Reply:
I wasn't including the question player since there isn't a three-dot menu in practice player's navbar, but it is available in the revision card (mentioned in [this thread](https://docs.google.com/document/d/1_RTzad6u4m-ZFn0VTWNddp4bc0fCYrMk-ABIVYVJ7gU/edit?pli=1&disco=AAAALqa4y7k)). If the question player is going to have the menu included in it soon then I agree that focusing on just the exploration player and hamburger menu would be more feasible; but if not then I think it's more sensible to include it in the revision card for homogeneity since it's odd to have the two three-dot menu have different selection options.